gatk_collect_insert_size_metrics_4.1.8.0/gatk_collect_insert_size_metrics_4.1.8.0.cwl

```diff
@@ -0,0 +1,200 @@
+  - id: assume_sorted
+    default: true
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: --ASSUME_SORTED
+    doc: >-
+      If true (default), then the sort order in the header file will be ignored.  Default value:
+      true. This option can be set to 'null' to clear the default value. Possible values: {true,
+      false}
+  - id: stop_after
+    type: int?
+    inputBinding:
+      position: 0
+      prefix: --STOP_AFTER
+    doc: >-
+      Stop after processing N reads, mainly for debugging. Default value: 0.
+      This option can be set to 'null' to clear the default value.
+  - id: create_index
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: --CREATE_INDEX
+    doc: >-
+      Whether to create a BAM index when writing a coordinate-sorted BAM file.  Default value:
+      false. Possible values: {true, false}
+  - id: create_md5_file
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: --CREATE_MD5_FILE
+    doc: >-
+      Whether to create an MD5 digest for any BAM or FASTQ files created.    Default value:
+      false. Possible values: {true, false}
+outputs:
+  - id: insert_size_metrics
+    type: File
+    outputBinding:
+      glob: |-
+        ${
+            if(inputs.output_file_name){
+                return inputs.output_file_name
+            } else {
+                return inputs.input.basename.replace(/.bam/, '_insert_size_metrics.txt')
+            }
+        }
+  - id: histogram_file_out
+    type: File
+    outputBinding:
+      glob: |-
+        ${
+            if(inputs.histogram_file){
+                return inputs.histogram_file
+            } else {
+                return inputs.input.basename.replace(/.bam/, '_histogram.pdf')
+            }
+        }
+arguments:
+  - position: 0
+    prefix: '--java-options'
+    valueFrom: |-
+      ${
+        if(inputs.memory_per_job && inputs.memory_overhead) {
+          if(inputs.memory_per_job % 1000 == 0) {
+            return "-Xmx" + (inputs.memory_per_job/1000).toString() + "G"
+          }
+          else {
+            return "-Xmx" + Math.floor((inputs.memory_per_job/1000)).toString() + "G"
+          }
+        }
+        else if (inputs.memory_per_job && !inputs.memory_overhead){
+          if(inputs.memory_per_job % 1000 == 0) {
+            return "-Xmx" + (inputs.memory_per_job/1000).toString() + "G"
+          }
+          else {
+            return "-Xmx" + Math.floor((inputs.memory_per_job/1000)).toString() + "G"
+          }
+        }
+        else if(!inputs.memory_per_job && inputs.memory_overhead){
+          return "-Xmx15G"
+        }
+        else {
+            return "-Xmx15G"
+        }
+      }
+  - position: 0
+    prefix: '--TMP_DIR'
+    valueFrom: .
+  - position: 0
+    prefix: '--COMPRESSION_LEVEL'
+    valueFrom: '2'
+  - position: 0
+    prefix: '--MAX_RECORDS_IN_RAM'
+    valueFrom: '50000'
+requirements:
+  - class: ResourceRequirement
+    ramMin: 32000
```

Comment:
32G is large but maybe this is a safe value to start

Reply:
@ionox0 heh yeah I just copied the base cwl from another tool and didn't bather changing it. i'll keep it as is then